src/types/StopLoss.sol

```diff
@@ -51,11 +53,33 @@ contract StopLoss is BaseConditionalOrder {
         returns (GPv2Order.Data memory order)
     {
         Data memory data = abi.decode(staticInput, (Data));
-        (, int256 latestSellPrice,,,) = data.sellTokenPriceOracle.latestRoundData();
-        (, int256 latestBuyPrice,,,) = data.buyTokenPriceOracle.latestRoundData();
+        // scope variables to avoid stack too deep error
+        {
+            (, int256 basePrice,,uint256 sellUpdatedAt,) = data.sellTokenPriceOracle.latestRoundData();
+            (, int256 quotePrice,,uint256 buyUpdatedAt,) = data.buyTokenPriceOracle.latestRoundData();
+
+            /// @dev Guard against invalid price data
+            if (!(basePrice > 0 && quotePrice > 0)) {
+                revert IConditionalOrder.OrderNotValid();
```

Comment:
Probably too late, but I wonder if OrderNotValid should have a string or something else as a parameter so we can specify more clearly which of the potential many conditions is not met (may make debugging easier).

Reply:
This is tracked in #32